staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -375,6 +378,13 @@ func (c *Config) Complete(informers informers.SharedInformerFactory) CompletedCo
 	if len(c.ExternalAddress) == 0 && c.PublicAddress != nil {
 		c.ExternalAddress = c.PublicAddress.String()
 	}
+	if c.PublicPort == 0 {
```

Comment:
This needs to be here or in test/integration when we start a master. Any preference?